src/Components/TestExplorer.fs

```diff
@@ -1525,19 +1988,108 @@ module Interactions =
 
                 let successfullyBuiltRequests = buildResults |> List.choose id
 
-                let! _ =
-                    successfullyBuiltRequests
-                    |> (Promise.executeWithMaxParallel maxParallelTestProjects runTestProject)
+                if useLegacyDotnetCliIntegration then
+                    let! _ =
+                        successfullyBuiltRequests
+                        |> (Promise.executeWithMaxParallel maxParallelTestProjects runTestProject)
+
+                    testRun.``end`` ()
+                else
+                    do! runTests_WithLanguageServer mergeTestResultsToExplorer testController.items req testRun
+                    testRun.``end`` ()
+                    do! discoverTests_WithLangaugeServer testItemFactory testController.items tryGetLocation
```

Comment:
Test runs can add test to the explorer. but the test run will not provide a code location.
This makes for a confusing and bad experience.

I tried a number of alternatives, and this one felt the best. 
Running after the test run avoids the test run feeling sluggish, and the discovery still runs fast enough (even with thousands of tests) that it's generally done by the time a dev is ready for a new action.

[More information](https://github.com/ionide/ionide-vscode-fsharp/pull/2106/commits/9c97dae6ab0e91edef4d2d247b5e991e3a943296)

A key alternative here would be running test discovery whenever a the test project build artifact is updated.
I think that's more in line with what Visual Studio does, but was a bigger deviation from our existing workflow